dist/src/test/java/io/camunda/application/CamundaDockerIT.java

```diff
@@ -0,0 +1,128 @@
+            .withNetwork(Network.SHARED)
+            .withNetworkAliases(ELASTICSEARCH_NETWORK_ALIAS)
+            .withEnv("xpack.security.enabled", "false")
+            .withExposedPorts(ELASTICSEARCH_PORT);
+    elasticsearchContainer.start();
+    camundaContainer =
+        new GenericContainer<>(CAMUNDA_TEST_DOCKER_IMAGE)
+            .withExposedPorts(SERVER_PORT, MANAGEMENT_PORT, GATEWAY_GRPC_PORT)
+            .withNetwork(Network.SHARED)
+            .withNetworkAliases(CAMUNDA_NETWORK_ALIAS)
+            .waitingFor(
+                new HttpWaitStrategy()
+                    .forPort(MANAGEMENT_PORT)
+                    .forPath("/actuator/health")
+                    .withReadTimeout(Duration.ofSeconds(120)))
+            .withStartupTimeout(Duration.ofSeconds(300))
+            .withEnv(
+                "ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_CLASSNAME",
+                "io.camunda.zeebe.exporter.ElasticsearchExporter")
+            .withEnv("ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_ARGS_URL", elasticsearchUrl())
+            .withEnv("ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_ARGS_BULK_SIZE", "1")
+            .withEnv("CAMUNDA_TASKLIST_ELASTICSEARCH_URL", elasticsearchUrl())
+            .withEnv("CAMUNDA_TASKLIST_ZEEBEELASTICSEARCH_URL", elasticsearchUrl())
+            .withEnv("CAMUNDA_TASKLIST_ZEEBE_GATEWAYADDRESS", gatewayAddress())
+            .withEnv("CAMUNDA_TASKLIST_ZEEBE_RESTADDRESS", httpUrl())
+            .withEnv("CAMUNDA_OPERATE_ELASTICSEARCH_URL", elasticsearchUrl())
+            .withEnv("CAMUNDA_OPERATE_ZEEBEELASTICSEARCH_URL", elasticsearchUrl())
+            .withEnv("CAMUNDA_OPERATE_ZEEBE_GATEWAYADDRESS", gatewayAddress());
+    try {
+      camundaContainer.start();
+    } catch (Exception e) {
+      fail(
+          String.format(
+              "Failed to start Camunda container.\n"
+                  + "Exception message: %s.\n"
+                  + "Container Logs:\n%s",
+              e.getMessage(), camundaContainer.getLogs()));
+    }
+    camundaContainer.start();
+
+    try (final CloseableHttpResponse response =
+        HttpClients.createDefault()
+            .execute(
+                new HttpGet(
+                    String.format(
+                        "http://%s:%d%s",
+                        camundaContainer.getHost(),
+                        camundaContainer.getMappedPort(MANAGEMENT_PORT),
+                        "/actuator/health")))) {
+      assertThat(response.getCode()).isEqualTo(200);
+      assertThat(EntityUtils.toString(response.getEntity()))
+          .isEqualTo(
+              "{\"status\":\"UP\",\"components\":{\"brokerReady\":{\"status\":\"UP\"},\"brokerStartup\":{\"status\":\"UP\"},\"brokerStatus\":{\"status\":\"UP\"},\"indicesCheck\":{\"status\":\"UP\"},\"livenessState\":{\"status\":\"UP\"},\"readinessState\":{\"status\":\"UP\"},\"searchEngineCheck\":{\"status\":\"UP\"}},\"groups\":[\"liveness\",\"readiness\",\"startup\",\"status\"]}");
+    }
+  }
+
```

Comment:
I would then propose like having the structure of:
```java
// given

// when
camunda.start

// then
checks for health status
```

❌ Furthermore, please make sure to use Awaitility for such cases, otherwise this test becomes easily flaky.

Reply:
+1 for the test structure, I will make sure to do that
regarding `Awaitility` use:
- for container start, I already use `HttpWaitStrategy` that regularly checks that the container health check before marking the container as started.
- for the health check response verification afterward, the health check should be available right away after the container is marked as started, so I don't expect that a retry would reduce the failure odds.
`Awaitility` is useful but IMO has no added value in this particular test.

Reply:
fixed in https://github.com/camunda/camunda/pull/19623/commits/ecd6e586376a0d2feedd0bc87cf4fbc31594a4b8